src/coreclr/jit/lowerxarch.cpp

```diff
@@ -1214,6 +1286,22 @@ GenTree* Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
             break;
         }
 
+        case NI_SSE2_Or:
+        case NI_AVX2_Or:
+        case NI_AVX512F_Or:
+        case NI_SSE2_Xor:
+        case NI_AVX2_Xor:
+        case NI_AVX512F_Xor:
+        {
+            GenTree* ternLogic = RecognizeHWTernaryLog(node);
+            if (ternLogic != nullptr)
+            {
+                // Revisit this node again.
+                return ternLogic->gtPrev;
+            }
+            break;
+        }
```

Comment:
What's the primary motivation for it being in lowering rather than morph? Most of our pattern based transform are recognized in `morph` and done there instead (including ROL/ROR, AND_NOT, etc).

Reply:
@tannergooding mainly, two reasons:
1) Handle LowerMemCmp which is also in lower
2) Handle user patterns with explicit intrinsics (since at this point we'll lower all cross-plat helpers to direct intrinsics)

Reply:
For 1, it seems like it'd be easier to just special case it. That's what we do for most of our other patterns as well. It might be even better if we had a way to share most of the necessary logic between morph/lowering, that way we can get the early ones (which should also help TP) and also avoid missing any of the late ones.

For 2, I don't see how that's impacted. We'll be recognizing the explicit intrinsics either way whether in morph or lowering, right?